templates/search_results.hbs

```diff
@@ -5,103 +5,102 @@
-  </header>
-
   <div class="search-results">
+    <section class="search-results-sidebar">
+      {{#if help_center_filters}}
+        <section class="filters-in-section collapsible-sidebar">
+          <h3 class="collapsible-sidebar-title sidenav-title">{{t 'filter_by_help_center'}}</h3>
+          <ul>
+            {{#each help_center_filters}}
+              <li>
+                <a href="{{url}}" class="sidenav-item" aria-selected="{{selected}}">{{name}} ({{count}})</a>
+              </li>
+            {{/each}}
+          </ul>
+        </section>
+      {{/if}}
+      {{#if help_center.community_enabled}}
+        <section class="filters-in-section collapsible-sidebar">
+          <h3 class="collapsible-sidebar-title sidenav-title">{{t 'filter_by_type'}}</h3>
+          <ul>
+            {{#each filters}}
+              <li>
+                <a href="{{url}}" class="sidenav-item" aria-selected="{{selected}}">{{name}} ({{count}})</a>
+              </li>
+            {{/each}}
+          </ul>
+        </section>
+      {{/if}}
+      {{#if subfilters}}
+        <section class="filters-in-section collapsible-sidebar">
+          {{#is current_filter.identifier 'knowledge_base'}}
+            <h3 class="collapsible-sidebar-title sidenav-title">{{t 'filter_by_category'}}</h3>
+          {{/is}}
+          {{#is current_filter.identifier 'community'}}
+            <h3 class="collapsible-sidebar-title sidenav-title">{{t 'filter_by_topic'}}</h3>
+          {{/is}}
+          <ul>
+            {{#each subfilters}}
+              <li>
+                <a href="{{url}}" class="sidenav-item" aria-selected="{{selected}}">{{name}} ({{count}})</a>
+              </li>
+            {{/each}}
+          </ul>
+        </section>
+      {{/if}}
+    </section>
+
     <section class="search-results-column">
-      <h2 class="search-results-subheading">
-        {{t 'knowledge_base'}}
-      </h2>
-      {{#if article_results}}
+      <h3 class="results-subheading">
+        {{#is current_filter.identifier 'unified'}}
+          {{t 'results' query=query count=results_count}}
+        {{else}}
+          {{#unless current_subfilter.identifier}}
+            {{t 'results' query=query count=results_count}}
+          {{else}}
+            {{t 'results_with_scope' query=query count=results_count scope_name=current_subfilter.name}}
+          {{/unless}}
+        {{/is}}
+      </h3>
+      {{#if results}}
         <ul class="search-results-list">
-          {{#each article_results}}
-            <li class="search-result">
-              <a href="{{url}}" class="search-result-link">{{title}}</a>
-              {{#if vote_sum}}
-                <span class="search-result-votes meta-count">{{vote_sum}}</span>
-              {{/if}}
+          {{#each results}}
+            <li class="search-results-list-item result-{{type}}">
+              <a href="{{url}}" class="results-list-item-link">{{title}}</a>
```

Comment:
Nit: For a11y, the titles should be encapsulated in a heading, e.g. `h2` to establish hierarchy. And each result could be wrapped in a `article` tag.